windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -148,51 +149,50 @@ func TestReceiveCommandStartStop(t *testing.T) {
+		"With no checksum file":             {missingChecksums: true, sendRootfsURL: "goodfile", wantInstalled: true},
 
-		"Error when the distroname is empty":         {emptyDistroName: true},
+		"Error when the distroname is empty":         {distroName: "-"},
 		"Error when the Appx does not exist":         {appxDoesNotExist: true},
 		"Error when the distro is already installed": {distroAlreadyInstalled: true, wantInstalled: true},
 		"Error when the distro fails to install":     {wslInstallErr: true},
 		"Error when cannot write cloud-init file":    {cloudInitWriteErr: true, wantCouldInitWriteCalled: true},
 
-		"Error when the rootfs isn't a valid tarball":                   {sendRootfsURL: "badfile", sendRootfsChecksum: mockErrorChecksum, wantInstalled: false},
-		"Error when the checksum doesn't match":                         {sendRootfsURL: "goodfile", sendRootfsChecksum: mockMismatchChecksum, wantInstalled: false},
-		"Error when the rootfs doesn't exist":                           {sendRootfsURL: "badresponse", wantInstalled: false},
-		"Error when URL doesn't respond":                                {sendRootfsURL: "goodfile", nonResponsiveServer: true, wantInstalled: false},
-		"Error when the destination dir for the VHDX cannot be created": {sendRootfsURL: "goodfile", breakVhdxDir: true, wantInstalled: false},
-		"Error when the rootfs tarball cannot be created":               {sendRootfsURL: "goodfile", breakTarFile: true, wantInstalled: false},
-		"Error when the rootfs temporary dir cannot be created":         {sendRootfsURL: "goodfile", breakTempDir: true, wantInstalled: false},
+		"Error when the distro ID is reserved (Ubuntu)":                   {sendRootfsURL: "goodfile", distroName: "Ubuntu", wantInstalled: false},
+		"Error when the distro ID is reserved (Preview)":                  {sendRootfsURL: "goodfile", distroName: "Ubuntu-Preview", wantInstalled: false},
+		"Error when the distro ID is reserved (case sensitiveness)":       {sendRootfsURL: "goodfile", distroName: "ubuntu-preview", wantInstalled: false},
+		"Error when the distro ID is reserved (release numbers)":          {sendRootfsURL: "goodfile", distroName: "Ubuntu-19.13", wantInstalled: false},
+		"Error when the distro ID is reserved (release numbers and case)": {sendRootfsURL: "goodfile", distroName: "uBuntu-19.13", wantInstalled: false},
+		"Error when the rootfs isn't a valid tarball":                     {sendRootfsURL: "badfile", wantInstalled: false},
+		"Error when the checksum doesn't match":                           {sendRootfsURL: "badchecksum", wantInstalled: false},
+		"Error when the checksum is missing for the rootfs":               {sendRootfsURL: "rootfswithnochecksum", wantInstalled: false},
+		"Error when the rootfs doesn't exist":                             {sendRootfsURL: "badresponse", wantInstalled: false},
+		"Error when the rootfs URL is ill-formed":                         {breakURL: true, sendRootfsURL: "goodfile", wantInstalled: false},
```

Comment:
I find it puzzling you are breaking the url while stil using a `sendRootfsURL` parameter in the same test. Maybe you can just use a hardcoded value like `@:?/BROKEN_URL`, and test this to not `url.JoinPath` in the test and take it verbatim?